static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js

```diff
@@ -17,75 +15,91 @@ define([
-                membershipLink: this.config.membershipLink,
-                signupDataLink: this.config.signupDataLink,
-                membershipDataLink: this.config.membershipDataLink,
-                subscriberLink: this.config.subscriberLink,
-                subscriberText: this.config.subscriberText,
-                subscriberDataLink: this.config.subscriberDataLink,
-                contributorLink: this.config.contributorLink,
-                contributorText: this.config.contributorText,
-                contributorDataLink: this.config.contributorDataLink,
-                showCloseBtn: this.config.showCloseBtn,
                 arrowWhiteRight: svgs('arrowWhiteRight'),
                 marque36icon: svgs('marque36icon'),
                 crossIcon: svgs('crossIcon'),
-                surveyOverlaySimple: svgs('surveyOverlaySimple')
+                surveyOverlaySimple: svgs('surveyOverlaySimple'),
+                adFreeDataLink: this.config.adFreeDataLink,
+                adFreeButtonText: this.config.adFreeButtonText,
+                adFreeMessagePrefix: this.config.adFreeMessagePrefix,
+                whitelistDataLink: this.config.whitelistDataLink,
+                whitelistText: this.config.whitelistText,
+                whitelistGuideImage: this.config.whitelistGuideImage,
+                adBlockIcon: this.config.adBlockIcon,
+                adBlockPlusIcon: this.config.adBlockPlusIcon,
+                variant: this.config.variant
             });
     };
 
     surveyAdBlock.prototype.attach = function () {
         fastdom.write(function () {
             $(document.body).append(this.bannerTmpl);
-            if (this.config.showCloseBtn) {
-                bean.on(document, 'click', $('.js-survey-adblock__close-btn'), function () {
-                    $('.survey-adblock').addClass('is-hidden');
-                    var cookieName = 'gu_abm_x',
-                        cookieLifetimeMinutes = 30,
-                        cookieCount = cookies.get(cookieName) ? parseInt(cookies.get(cookieName)) : 0;
-                    cookies.addForMinutes(cookieName, cookieCount + 1, cookieLifetimeMinutes);
-                });
-            }
+            bean.on(document, 'click', $('.cta-whitelist'), function () {
+                // -> state 2
+                $.forEachElement(('.state-1'), function(el){el.classList.add('is-hidden');});
+                $.forEachElement(('.state-3'), function(el){el.classList.add('is-hidden');});
+                $.forEachElement(('.state-4'), function(el){el.classList.add('is-hidden');});
+                $.forEachElement(('.state-2'), function(el){el.classList.remove('is-hidden');});
+                $('.survey-container').removeClass('thank-you');
+            });
+            bean.on(document, 'click', $('.survey-button-rounded__cta.noads'), function () {
+                // -> state 3
+                $.forEachElement(('.state-1'), function(el){el.classList.add('is-hidden');});
```

Comment:
Could you abstract out these show/hide functions, there seems to be a lot of duplication